### PR TITLE
feat(services): add NewRelic service

### PR DESCRIPTION
**Pull request by saviogl**
Hi folks,

My company is in the process of rolling out New Relic internally and I saw an opportunity for adding a new notification service for it to integrate with its [Deployment Marker](https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-monitor-deployments/) functionality.

Not sure if the contributing process involving opening up an issue first, etc, so just let me know if changes are need to be made.

Also, pretty new to the Argoproj codebase, so let me know if I missed any aspect of adding a new notification service, or if there are any other considerations/suggestions. Haven't tested e2e, not sure the best way of doing so within this codebase or integrated with the ArgoCD one.

Issue: #93

### CODE REVIEW

**Comment by saviogl** [comment]
@alexmt Just making sure I'm following the right approach here

**Comment by saviogl** [comment]
@ryota-sakamoto Maybe you can shed some light on the contributing process here?

**Comment by pasha-codefresh** [comment]
Hey @saviogl , i am going to review it

**Comment by saviogl** [comment]
@pasha-codefresh Thanks for getting back to me, I wasn't sure if this repository was still active, or how I could follow up on the contribution process. I had left a few messages back in the day on Slack, but to no avail.

Glad that we are coming around to it, it's been a while so I might need a refresher myself. I ended up using Webhooks Notifications Service to setup the foundation for the company that I worked for but it would be great to have this abstraction as a supported service.

Let me know if you have any questions...

On another note, If I wanted to get more involved with the argo project, where would be the best place to start?

**Comment by pasha-codefresh** [comment]
ArgoCD is good candidate, but notification is also part of it, i would say that notification also great place for beginning. 

About PR, the code itself looks good to me, I just want run a manual test on it today, and after we can merge

File: docs/services/newrelic.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+1. Create a NewRelic [Api Key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#user-api-key)
+2. Store apiKey in `argocd-notifications-secret` Secret and configure NewRelic integration in `argocd-notifications-cm` ConfigMap
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <config-map-name>
+data:
+  service.mattermost: |
```
Comment (pasha-codefresh) [inline docs/services/newrelic.md:19]: Please change it to newrelic

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: docs/services/newrelic.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  annotations:
+    notifications.argoproj.io/subscribe.<trigger-name>.newrelic: <app-id>
+```
+
+## Templates
```
Comment (pasha-codefresh) [inline docs/services/newrelic.md:44]: Probably also add description that cover it. ? 

```
commitAuthorTemplate := n.User
	if commitAuthorTemplate == "" {
		commitAuthorTemplate = "{{(call .repo.GetCommitMetadata .app.status.sync.revision).Author}}"
	}
```

You can hardcode user if needed

File: docs/services/newrelic.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+```yaml
+context: |
+  argocdUrl: https://example.com/argocd
+
+template.app-deployed: |
+  message: Application {{.app.metadata.name}} has been healthy.
+  newrelic:
+    description: Application {{.app.metadata.name}} has been healthy
+    changelog: "{{(call .repo.GetCommitMetadata .app.status.sync.revision).Message}}"
```
Comment (pasha-codefresh) [inline docs/services/newrelic.md:57]: If you not define user you take it from metadata, probably same should be done for changelog ?

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

**Comment by pasha-codefresh** [comment]
@saviogl i got it working, great job, lets just fix small issues in documentation 
Снимок экрана 2022-11-27 в 19 34 01

**Comment by pasha-codefresh** [comment]
Hey @saviogl , do you need some help with solve comments that i provided?

**Comment by saviogl** [comment]
Hi @pasha-codefresh, I'll be looking at this today, and I'll let you know if I come across any questions? Are you in CNCF Slack workspace?

**Comment by pasha-codefresh** [comment]
Yep @saviogl , messaged you in slack

**Review by pasha-codefresh — APPROVED** [review]

**Comment by pasha-codefresh** [comment]
Great job, thank you!

**Pull request merged**